tvaultuiv2/src/views/private/service-accounts/components/ServiceAccountPermission/components/AppRoles/index.js

```diff
@@ -0,0 +1,289 @@
+  useEffect(() => {
+    if (newAppRole) {
+      setResponse({ status: 'add' });
+    }
+  }, [newAppRole]);
+
+  const constructPayload = (role, access) => {
+    const data = {
+      access: checkAccess(access),
+      approlename: role,
+      svcAccName: accountDetail.name,
+    };
+    return data;
+  };
+
+  /**
+   * @function onDeleteClick
+   * @description function to delete the app role from the svc account app role list.
+   * @param {role} string app role name.
+   * @param {access} string permission of the app role.
+   */
+  const onDeleteClick = (role, access) => {
+    setResponse({ status: 'loading' });
+    const payload = constructPayload(role, access);
+    apiService
+      .deleteAppRolePermission(payload)
+      .then(async (res) => {
+        if (res && res.data?.messages && res.data?.messages[0]) {
+          updateToastMessage(1, res.data.messages[0]);
+          setResponse({ status: '' });
+          await fetchPermission();
+        }
+      })
+      .catch((err) => {
+        setResponse({ status: 'success' });
+        if (err.response?.data?.errors && err.response.data.errors[0]) {
+          updateToastMessage(-1, err.response.data.errors[0]);
+        }
+      });
+  };
+
+  /**
+   * @function onSaveClicked
+   * @description function to save the app rolr to the svc account app role list.
+   * @param {data} object payload to call api.
+   */
+  const onSaveClicked = (data) => {
+    setResponse({ status: 'loading' });
+    apiService
+      .addAppRolePermission(data)
+      .then(async (res) => {
+        if (res?.data?.messages && res.data?.messages[0]) {
+          updateToastMessage(1, res.data?.messages[0]);
+          setResponse({ status: '' });
+          await fetchPermission();
+        }
+      })
+      .catch((err) => {
+        if (err.response?.data?.errors && err.response.data.errors[0]) {
+          updateToastMessage(-1, err.response.data.errors[0]);
+        }
+        setResponse({ status: 'success' });
+      });
+  };
+
+  /**
+   * @function onSubmit
+   * @description function structure the payload when save/edit is clicked and call save api.
+   * @param {role} string role name.
+   * @param {access} string permission given to the app role.
+   */
+  const onSubmit = async (role, access) => {
+    const payload = constructPayload(role, access);
+    await onSaveClicked(payload);
+    onNewAppRoleChange();
+  };
+
+  /**
+   * @function onEditSaveClicked
+   * @description function to edit the existing app role.
+   * @param {role} string app role name to edit.
+   * @param {access} string permission given to the app role.
+   */
+  const onEditSaveClicked = (role, access) => {
+    setResponse({ status: 'loading' });
+    const payload = constructPayload(role, access);
+    apiService
+      .deleteAppRolePermission(payload)
+      .then((res) => {
+        if (res) {
+          setResponse({ status: 'loading' });
+          onSubmit(role, access);
```

Comment:
onSubmit returns a promise?? Should  be awaited

Reply:
addressed